typescript/009-matrix/matrix.ts

```diff
@@ -0,0 +1,17 @@
+export class Matrix {
+  readonly matrix: number[][];
+
+  constructor(input: string) {
+    this.matrix = input.split('\n').map(row =>
+      row.split(/\s+/).map(n => Number(n))
+    );
+  }
+
+  get rows(): number[][] {
+    return this.matrix;
+  }
+
+  get columns(): number[][] {
+    return this.matrix.map((_, col) => this.matrix.map(row => row[col]));
```

Comment:
1) It looks like the row is indexed by column which makes no sense from type's perspective (array indexed by array). Name variables for what they are.
2) `this.matrix.map((_, col) =>` is semantically incorrect - you're not iterating over columns, so why is it `col`?